File: golang/49_group_anagrams.go
package main

import (
	"fmt"
	"sort"
	"strings"
)

func groupAnagrams(strs []string) [][]string {
	groupMap := make(map[string]string)
	for i := 0; i < len(strs); i++ {
		temp := strings.Split(strs[i], "")
		sort.Strings(temp)
		sorted := strings.Join(temp, "")
		if _, ok := groupMap[sorted]; ok {
			groupMap[sorted] += ", " + strs[i]
		} else {
			groupMap[sorted] = strs[i]
		}
	}
	res := make([][]string, len(groupMap))
	var ind int = 0
	for _, v := range groupMap {
		res[ind] = append(strings.Split(v, ", "))
		ind++
	}
	return res
}

func main() {
	var inp string
	fmt.Scanf("%s", &inp)
	fmt.Println("innnnn::: ", inp)
	inpList := strings.Split(inp, ", ")
	fmt.Println("inoutList:: ", inpList)
	fmt.Println(groupAnagrams(inpList))
}

File: python/49_group_anagrams.py
from typing import List

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        group_anagrams = {}
        for strings in strs:
            key_val = ''.join(sorted(strings))
            # print(key_val)
            if key_val in group_anagrams:
                group_anagrams[key_val].insert(len(group_anagrams[key_val]), strings)
            else:
                group_anagrams[key_val] = [strings]
        res = []
        for keys in group_anagrams:
            res.insert(len(res), group_anagrams[keys])
        return res

sol = Solution()
inp = input().split(", ")
print(sol.groupAnagrams(inp))